projects/plugins/jetpack/extensions/blocks/blogroll/use-site-recommendations.js

```diff
@@ -0,0 +1,79 @@
+import { store as coreStore, useEntityProp } from '@wordpress/core-data';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState, useMemo } from '@wordpress/element';
+import { debounce } from '../../shared/debounce';
+import { createBlockFromRecommendation } from './utils';
+
+export function useBlogrollItemAttributes( clientId ) {
+	const block = useSelect(
+		select => {
+			return select( 'core/block-editor' ).getBlock( clientId );
+		},
+		[ clientId ]
+	);
+
+	return useMemo( () => {
+		return block?.innerBlocks
+			.filter( ( { name } ) => name === 'jetpack/blogroll-item' )
+			.map( ( { attributes } ) => {
+				return attributes;
+			} );
```

Comment:
I don't think you should `useMemo` honestly. They're under 100 items, that's absolutely nothing to a computer.